### PR TITLE
WIP: Bring odroid-c2 back up to date

### PR DESCRIPTION
**Pull request by bspeice**
Hoping to get this running again, I'd be interested in using it for some personal work.

### CODE REVIEW

**Comment by StefanScherer** [comment]
@bspeice I love it. Thanks. I don't have Odroid devices so I cannot help test, but can give you advise with Circle etc.

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+              ./ghr $pre -u hypriot $CIRCLE_TAG output/
+            fi
+workflows:
+  version: 2
+  build_and_deploy:
+    jobs:
+      - build:
+          filters:
+            tags:
+only: /.*/
```
Comment (StefanScherer) [inline .circleci/config.yml]: Check indentation. See https://github.com/hypriot/image-builder-rpi/blob/89a69c3977d9dca1d2f55fc5355bf20bfa127110/.circleci/config.yml#L46 for reference.

**Comment by StefanScherer** [comment]
I have activated Circle builds, so you should be able to push to your PR branch.

File: .circleci/config.yml (hunk 1 above)
Comment (bspeice) [reply .circleci/config.yml]: That explains the config errors on CircleCI, thanks.

**Comment by bspeice** [comment]
I'm expecting testing to be the easy part, I'm still working on getting the full build working. I've been having issues with Guestfish, but I'm successful up until that point.

**Comment by bspeice** [comment]
@StefanScherer - Something weird is going on with Guestfish, and the drive is filling up. If you get a chance, I could use some help.
Alternately, would it be worth skipping the Guestfish section (and use dd instead)? I'm not sure Qemu/Guestfish is actually needed if the end result is the same as the sd_fuse putting the bootloader into the first partition.

**Comment by StefanScherer** [comment]
@bspeice You could skip Guestfish and use other tools instead. We decided to use Guestfish as it looked simpler, but it has some hard constraints (eg. does not work in D4M).

**Comment by bspeice** [comment]
Publishing a fixed build momentarily, but it looks like there was a bug in `supermin` which caused guestfish to fail: https://bugzilla.redhat.com/show_bug.cgi?id=1532978

Adding stretch-backports and installing the newest supermin does the trick.

EDIT: That's probably something that should be added to image-builder ultimately. In the meantime, going to just add for `build.sh` to get this passing.

**Comment by StefanScherer** [comment]
I have activated pull request builds for forks, I missed that last week.

**Comment by bspeice** [comment]
Seems that it's not testing anything - I copied the circleci config from the rpi image, not sure what's going on there.

**Comment by StefanScherer** [comment]
Hm, build 5 and 6 fall back to Circle 1.0 https://circleci.com/gh/bspeice/image-builder-odroid-c2/tree/master
Seems like a problem in the .circleci/config.yml

**Comment by StefanScherer** [comment]
Found it: Rename  .circleci/config.yaml to  .circleci/config.yml without an "a".

**Comment by StefanScherer** [comment]
Could you update the build badge in the README.md as well?

```
[![CircleCI](https://circleci.com/gh/hypriot/image-builder-odroid-c2.svg?style=svg)](https://circleci.com/gh/hypriot/image-builder-odroid-c2)
```

**Comment by bspeice** [comment]
Also, note that while I'm expecting the build to succeed, it's not going to be ready for merge yet. I still need to take it for a spin on the actual board, and cloud-init isn't currently installed.

**Comment by bspeice** [comment]
```
[bspeice@bspeice-MS-7971: image-builder-odroid-c2]$ ssh pirate@192.168.1.103
The authenticity of host '192.168.1.103 (192.168.1.103)' can't be established.
ECDSA key fingerprint is SHA256:dl9rJeNVnb5ssuShQ1hi6WdTn1KGpFcM1Nqtc6adVBw.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '192.168.1.103' (ECDSA) to the list of known hosts.
pirate@192.168.1.103's password:
Linux black-pearl 3.14.79-117 #1 SMP PREEMPT Tue Jan 2 23:46:30 BRST 2018 aarch64

HypriotOS (Debian GNU/Linux 9)

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
HypriotOS/arm64: pirate@black-pearl in ~
$ ls
HypriotOS/arm64: pirate@black-pearl in ~
$ docker run hello-world
Unable to find image 'hello-world:latest' locally
latest: Pulling from library/hello-world
255483503861: Pull complete
Digest: sha256:f5233545e43561214ca4891fd1157e1c3c563316ed8e237750d59bde73361e77
Status: Downloaded newer image for hello-world:latest

Hello from Docker!
This message shows that your installation appears to be working correctly.

To generate this message, Docker took the following steps:
 1. The Docker client contacted the Docker daemon.
 2. The Docker daemon pulled the "hello-world" image from the Docker Hub.
    (arm64v8)
 3. The Docker daemon created a new container from that image which runs the
    executable that produces the output you are currently reading.
 4. The Docker daemon streamed that output to the Docker client, which sent it
    to your terminal.

To try something more ambitious, you can run an Ubuntu container with:
 $ docker run -it ubuntu bash

Share images, automate workflows, and more with a free Docker ID:
 https://hub.docker.com/

For more examples and ideas, visit:
 https://docs.docker.com/engine/userguide/
```

Can also close #30, I've included cloud-init on this image.

**Comment by StefanScherer** [comment]
This is fantastic @bspeice
I‘ll merge it and draft a pre-release to check the whole pipeline.

**Comment by StefanScherer** [comment]
There's a new release v0.4.0: https://github.com/hypriot/image-builder-odroid-c2/releases/tag/v0.4.0

**Pull request merged**